### DISCUSSION TITLE
Supports filter string using regular expression matching

### DISCUSSION THREAD

**Discussion by chai3** [langfuse/langfuse#3014] [Ideas]
### Describe the feature or potential improvement

regex is powerful

![image](https://github.com/user-attachments/assets/efe18557-ff00-4ab0-955a-20f351a63191)

### Additional information

Probably add 3 lines

https://github.com/langfuse/langfuse/blob/bb6bc79bdceabbb2b664b31f30e090dd4849921a/packages/shared/src/interfaces/filters.ts#L5
https://github.com/langfuse/langfuse/blob/bb6bc79bdceabbb2b664b31f30e090dd4849921a/packages/shared/src/interfaces/filters.ts#L9-L15
add `"regex with"`

https://github.com/langfuse/langfuse/blob/bb6bc79bdceabbb2b664b31f30e090dd4849921a/packages/shared/src/filterToPrisma.ts#L6-L13
add `"regex with": "~*"` // case-insensitively
or `"~"` // case sensitively

We can add `"does not regex with": "!~*"` or `!~`

see also https://www.postgresql.org/docs/current/functions-matching.html#FUNCTIONS-POSIX-REGEXP

**Comment by marcklingen** [comment]
thanks for sharing, great idea! are you interested in contributing this?

**Reply by chai3** [reply]
@marcklingen YES

**Reply by chai3** [reply]
@marcklingen I created https://github.com/langfuse/langfuse/pull/3055
